web/src/topic_list.ts

```diff
@@ -499,10 +482,22 @@ export function get_left_sidebar_topic_search_term(): string {
     return $("#topic_filter_query").text().trim();
 }
 
-export function get_typeahead_search_term(): string {
+export function get_typeahead_search_pills_syntax(): string {
     const $pills = $("#left-sidebar-filter-topic-input .pill");
-    const value = $pills.find(".pill-value").text().trim();
-    return value;
+    if ($pills.length === 0) {
+        return "";
+    }
+    // For now, there is only one pill in the left sidebar topic search input.
+    // This is because we only allow one topic filter pill at a time.
+    // If we allow multiple pills in the future, we may need to
+    // change this logic to return the syntax of all pills.
+    if ($pills.length > 1) {
+        blueslip.warn("Multiple pills found in left sidebar topic search input.");
+    }
+    // We can remove this assumption once we allow multiple pills and hence update the
+    // callers of this function to handle multiple pills and implement the search accordingly.
+    const syntax = $pills.attr("data-syntax");
+    return syntax ?? "";
 }
```

Comment:
I think a better way to handle this would be to query `topic_filter_pill_widget.items()` similar to the `get_current_search_pill_terms` method in `web/src/search_pill.ts`.

Reply:
This should also remove the need for an additional `data-syntax` attribute in the pill template? CC @timabbott for your thoughts, since the issue description specifically mentions the need for a `data-syntax` attribute.

Reply:
Yeah it's possible the `data-syntax` attribute have been a workaround that may not be required after this refactor.